src/backend/mpi/swdsm.cpp

```diff
@@ -541,291 +529,251 @@ void write_back_writebuffer() {
 }
 
 void load_cache_entry(unsigned long loadtag, unsigned long loadline) {
-	int i;
-	unsigned long homenode;
-	unsigned long id = 1 << getID();
-	unsigned long invid = ~id;
 
-	if(loadtag>=size_of_all){//Trying to access/prefetch out of memory
-		return;
+	if(loadtag >= size_of_all){ // Address out of bounds
+		printf("Address %lu out of bounds.\n", loadtag);
+		exit(EXIT_FAILURE);
 	}
```

Comment:
This should probably return instead of exit, as a segfault outside of the ArgoDSM memory range should not be handled by ArgoDSM. However, it is unclear if `load_cache_entry` can be called with an invalid "tag"/address in the first place.